### DISCUSSION TITLE
Help with downloading and using 'youtube-dl'

### DISCUSSION THREAD

**Discussion by SexyCurly** [community/community#139360] [New to GitHub]
I don't understand github. I don't know what I'm supposed to do or what or how I download anything or how I get it to work once it's downloaded. I don't understand 'Terminal'. Would someone please give me straightforward step by step instructions on how I download 'youtube-dl' and how I get it to work. I'm not a developer so I don't understand anything! I just want to be able to download this program and use it to download and save some YouTube videos. Please help! I'm currently running 'MacOS Catalina 10.15.7

**Comment by sam-math** [comment]
1. **Install Homebrew** (if you don’t have it yet):
   - Homebrew is a package manager that makes it easy to install things like `youtube-dl`. 
   - Open your Terminal (you can find it by searching for “Terminal” in Spotlight).
   - Copy and paste this command, then press **Enter**:
     ```
     /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
     ```
   - Follow the on-screen instructions to install Homebrew. It may take a few minutes.

2. **Install youtube-dl**:
   - Once Homebrew is installed, type this command in your Terminal and hit **Enter**:
     ```
     brew install youtube-dl
     ```

3. **Download a YouTube video**:
   - After it installs, you can download YouTube videos. Just type the following command in Terminal:
     ```
     youtube-dl <paste-your-youtube-link-here>
     ```
   - Replace `<paste-your-youtube-link-here>` with the actual URL of the YouTube video you want to download. Press **Enter**, and it will start downloading the video to your current folder (usually your home directory).

4. **Find the downloaded video**:
   - By default, it will save the video to the folder where you're running Terminal. You can use Finder to go to your Home folder to find it, or you can use Terminal to specify where to save the video like this:
     ```
     youtube-dl -o ~/Downloads/%(title)s.%(ext)s <paste-your-youtube-link-here>
     ```
   - This will save the video directly to your **Downloads** folder.

That’s it!

**Comment by SexyCurly** [comment]
Thank you for your helpful instructions. They seem clear enough to follow and I’ll see how I go. If I hit any problems I’ll let you know.

- Edward -

**Reply by sam-math** [reply]
Happy to help. Once you 've checked it works please mark my answer as answer (so it can help others find a solution faster in the future)
